### PR TITLE
fix(backup): Pull correct GCP credentials file path

### PR DESCRIPTION
**Pull request by azaslavsky**

### CODE REVIEW

File: src/sentry/utils/env.py [hunk 1]
```diff
@@ -14,12 +15,22 @@ def gcp_project_id() -> str:
...
-    adc_path = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS", "")
-    if adc_path:
-        with open(adc_path) as fp:
-            adc = json.load(fp)
-            if adc.get("quota_project_id") is not None:
-                return adc.get("quota_project_id")
+    # Try the metadata endpoint, if possible. If not, we'll assume a local environment as use the
+    # app credentials env variable instead.
+    try:
+        requests.get(
```
Comment (maxwelllwang) [inline src/sentry/utils/env.py:21]: where are you using the request response? should it return request.response

Comment (maxwelllwang) [reply src/sentry/utils/env.py:21]: but this seems like it will work both in single tenant and saas environments. just tried it

Comment (azaslavsky) [reply src/sentry/utils/env.py]: Force pushed a fix here.

**Review by maxwelllwang — APPROVED** [review]
Looks good, there was a misunderstanding for what using ADC path was for. It is used for local testing. Should we remove before completing this project?

**Comment by azaslavsky** [comment]
I don't think we want to remove it. The `gcp_project_id` method exists to abstract over 4 environments this code could be run in:

1. During tests.
2. On a local dev machine, when they have not set up their configuration to use GCP (no `GOOGLE_APPLICATION_CREDENTIALS`, no GCP filestore, etc)
3. On a local dev machine, when they HAVE set configured GCP.
4. In prod on GCP boxes.

We'll likely want to support all of these configurations in perpetuity going forward.

**Pull request merged**